pkg/marathon/client.go

```diff
@@ -160,3 +160,55 @@ func (c *Client) Queue() (marathon.Queue, error) {
 		return marathon.Queue{}, errors.New("unable to get Marathon queue")
 	}
 }
+
+func (c *Client) ApplicationVersions(appID string) (marathon.ApplicationVersions, error) {
+	dcosClient := pluginutil.HTTPClient(c.baseURL)
+	resp, err := dcosClient.Get(fmt.Sprintf("/v2/apps%s/versions", normalizeAppID(appID)))
+	if err != nil {
+		return marathon.ApplicationVersions{}, err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case 200:
+		var result marathon.ApplicationVersions
+		err = json.NewDecoder(resp.Body).Decode(&result)
+		return result, err
+	default:
+		return marathon.ApplicationVersions{}, fmt.Errorf("unable to get versions for Marathon app %s", appID)
+	}
+}
+
+func (c *Client) ApplicationByVersion(appID string, version string) (*marathon.Application, error) {
+	dcosClient := pluginutil.HTTPClient(c.baseURL)
```

Comment:
go-marathon has `ApplicationByVersion`. Why not use that?

Reply:
I thought we were mainly use go-marathon for its types rather than the API parts and also, `ApplicationByVersion` works differently (calls `/v2/apps/{}/versions/{}`) which we only want to do in certain circumstances. And this is a more direct porting of the Python code.